src/shared/sleep-config.c

```diff
@@ -385,24 +416,45 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
                         };
 
                         /* if the swap is the resume device, stop looping swaps */
-                        if (location_is_resume_device(hibernate_location, sys_resume, sys_offset))
+                        if (location_is_resume_device(hibernate_location, sys_resume, sys_offset) > 0)
                                 break;
                 }
         }
 
         if (!hibernate_location)
                 return log_debug_errno(SYNTHETIC_ERRNO(ENOSYS), "No swap partitions or files were found");
 
-        if (!streq(sys_resume, "0:0") && !location_is_resume_device(hibernate_location, sys_resume, sys_offset))
+        int resume_match = location_is_resume_device(hibernate_location, sys_resume, sys_offset);
```

Comment:
This will only check whether the selected swap entry (i.e. the "highest priority swap with most remaining space") is on a Btrfs file system. This fails in case
- there are multiple swap entries and
- the hibernation location is a Btrfs swap file (hence cannot be found by `location_is_resume_device` in the parsing loop), but
- the selected swap entry is not on a Btrfs file system.

Reply:
I think that would only happen if you had a swap file on a non-Btfs fs configured with a higher priority than the swap entry for `resume=` and `resume_offset=`.

I wonder if what we should do instead is introduce a new Environment variable here that would allow Btrfs swapfile users to skip the code that attempts to do location and just returns the `resume=` and `resume_offset=` values.

For example, if  `SYSTEMD_BYPASS_HIBERNATION_LOCATION_DETECTION` is set, `find_hibernate_location` would just return a `HibernateLocation` for the values read from `resume=` and `resume_offset=` before entering into the loop that tries to match the swaps.

Reply:
@diabonas, I pushed up a new change that will allow `SYSTEMD_BYPASS_HIBERNATION_MEMORY_CHECK` to bypass the location detection if the user has correctly configure `/sys/power/resume` (and by extension, `/sys/power/resume_offset` when using a swapfile).